src/wallet/rpcdump.cpp

```diff
@@ -805,35 +805,190 @@ UniValue dumpwallet(const JSONRPCRequest& request)
     return reply;
 }
 
+struct UsageMaps {
+    using ScriptUseMap = std::map<CScriptID, bool>;
+    using WatchKeyUseMap = std::map<CKeyID, bool>;
+    using KeyUseMap = std::map<CKeyID, bool>;
+
+    WatchKeyUseMap map_watch_key_use;
+    ScriptUseMap map_script_use;
+    KeyUseMap map_key_use;
+};
+
+// Class which remembers whether every script and public key was used
+class AllUsedKeyStore : public CBasicKeyStore
+{
+private:
+    struct UsageMaps* usage = new UsageMaps();
+public:
+    bool AllUsed() const;
+    bool GetPubKey(const CKeyID &address, CPubKey& vchPubKeyOut) const override;
```

Comment:
Make the parameters names match between declaration and definition (`CPubKey& vchPubKeyOut` vs `CPubKey &pubkey_out`) :-)